File: data/code_pairs/cpp/1653.cpp
#include <iostream>
#include <cmath>
#include <cassert>

double distance(double asep, double eccn, double Tanom) {
    return asep * (1.0 - eccn * eccn) / (1 + eccn * std::cos(Tanom));
}

double trueanomaly(double eccn, double Eanom) {
    double temp1 = std::sqrt((1.0 + eccn) / (1.0 - eccn));
    double temp2 = std::tan(Eanom / 2.0);
    return 2.0 * std::atan(temp1 * temp2);
}

void kepler(double Manom, double& Eanom, double eccn) {
    const int itmax = 100;
    double Eold, diff, thres = 1.0e-8;
    int i = 0;

    do {
        Eold = Eanom;
        Eanom = Manom + eccn * std::sin(Eold);
        diff = std::abs(Eanom - Eold);
        i++;
    } while (diff > thres && i < itmax);
}

void invkepler(double Eanom, double& Manom, double eccn) {
    const int itmax = 100;
    double Mold, diff, thres = 1.0e-6;
    int i = 0;

    do {
        Mold = Manom;
        Manom = Eanom - eccn * std::sin(Mold);
        diff = std::abs(Manom - Mold);
        i++;
    } while (diff > thres && i < itmax);
}

bool areClose(double a, double b, double tolerance = 1e-8) {
    return std::abs(a - b) < tolerance;
}

int main() {
    assert(areClose(distance(1.0, 0.1, 0.5), 0.91012869299513888));
    assert(areClose(trueanomaly(0.1, 0.5), 0.55026397471366328));
    
    double Eanom = 0.1;
    kepler(0.5, Eanom, 0.1);
    assert(areClose(Eanom, 0.55247998678778321));

    double Manom = 0.1;
    invkepler(0.5, Manom, 0.1);
    assert(areClose(Manom, 0.45596695102025764));

    std::cout << "All tests passed!" << std::endl;

    return 0;
}

File: data/code_pairs/fortran/1653.f90
module astronomy_module
  implicit none
contains

  double precision function distance(asep, eccn, Tanom)
    double precision, intent(in) :: asep, eccn, Tanom
    distance = asep * (1.0d0 - eccn * eccn) / (1 + eccn * cos(Tanom))
  end function distance

  double precision function trueanomaly(eccn, Eanom)
    double precision, intent(in) :: eccn, Eanom
    double precision :: temp1, temp2
    temp1 = sqrt((1.0d0 + eccn) / (1.0d0 - eccn))
    temp2 = tan(Eanom / 2.0d0)
    trueanomaly = 2.0d0 * atan(temp1 * temp2)
  end function trueanomaly

  subroutine kepler(Manom, Eanom, eccn)
    double precision, intent(in) :: Manom, eccn
    double precision, intent(inout) :: Eanom
    double precision :: Eold, diff, thres
    integer :: i, itmax
    thres = 1.0d-8
    itmax = 100
    i = 0

    do
      Eold = Eanom
      Eanom = Manom + eccn * sin(Eold)
      diff = abs(Eanom - Eold)
      if (diff <= thres .or. i >= itmax) exit
      i = i + 1
    end do
  end subroutine kepler

  subroutine invkepler(Eanom, Manom, eccn)
    double precision, intent(in) :: Eanom, eccn
    double precision, intent(inout) :: Manom
    double precision :: Mold, diff, thres
    integer :: i, itmax
    thres = 1.0d-6
    itmax = 100
    i = 0

    do
      Mold = Manom
      Manom = Eanom - eccn * sin(Mold)
      diff = abs(Manom - Mold)
      if (diff <= thres .or. i >= itmax) exit
      i = i + 1
    end do
  end subroutine invkepler

end module astronomy_module

program test_astronomy
  use astronomy_module
  implicit none
  print *, "Distance Test:", distance(1.0d0, 0.1d0, 0.5d0)
  print *, "True Anomaly Test:", trueanomaly(0.1d0, 0.5d0)
  ! Note: The values for Eanom in kepler and invkepler tests are initial guesses and may be modified as needed
  print *, "Kepler Test:", kepler(0.5d0, 0.1d0, 0.1d0)
  print *, "Inverse Kepler Test:", invkepler(0.5d0, 0.1d0, 0.1d0)
end program test_astronomy